pandas/core/base.py

```diff
@@ -497,6 +498,19 @@ def searchsorted(self, key, side='left'):
         #### needs tests/doc-string
         return self.values.searchsorted(key, side=side)
 
+    # string methods
+    def _make_str_accessor(self):
+        if not com.is_object_dtype(self.dtype):
```

Comment:
I think better to check the inferred type
which will be string or mixed for s multi index (iirc)